SpringProject/src/main/java/com/gradproject2019/users/controller/UserController.java

```diff
@@ -43,4 +43,10 @@
         UserResponseDto userResponseDto = userService.findUserById(userId, token);
         return ResponseEntity.ok(userResponseDto);
     }
-}
+
+    @GetMapping(path = "/users")
```

Comment:
OK, Looking at this again I realise I've been a darn fool 
There is no need to pass in the `manager_id` at all - you have that from the `token` (indirectly).

So in your business you should use the mapping/token table to retrieve the user/manager id, then build the query based on that.  

So the base query for get users should be;

```
SELECT * FROM users 
WHERE manager_id=:userId
OR user_id = :userId
```

If I understand it right the only users you can see when using the app, are the current user's employees? (and yourself). 

If you introduce admins/superusers you would need to remove this default filter

Then additional query params, like search name would just add new filters to that query.